jetty-server/src/main/java/org/eclipse/jetty/server/AsyncContentProducer.java

```diff
@@ -219,39 +219,43 @@ public void reclaim(HttpInput.Content content)
     public boolean isReady()
     {
         HttpInput.Content content = nextTransformedContent();
-        if (content == null)
+        if (content != null)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("isReady(), got transformed content {} {}", content, this);
+            _httpChannel.getState().onContentAdded();
```

Comment:
Can you add a TODO here to review if this is needed.... or @lorban  can you open an issue to check.  Eitherway, is unrelated to this PR

Reply:
I suspect there's already enough coverage to tell us if omitting the `onContentAdded()` call here would fail anything, but it's an experiment we ought to do. I've opened https://github.com/eclipse/jetty.project/issues/5704 to track this.